app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -175,7 +177,7 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         setupThemeAndBrowsingMode(getModeFromIntentOrLastKnown(intent))
         setContentView(R.layout.activity_home)
 
-        createNavGraph()
+        val navGraphJob = createNavGraph()
```

Comment:
Thinking aloud, I'm concerned we're going to lose track of all the async + "barrier" things we're doing... I wonder if we should put a light abstraction layer around this so we don't have to rummage through the code to remember all of them (e.g. put all the async stuff in one file).

Maybe this isn't needed yet but throwing it out there.